postgresql-dialect/src/commonMain/kotlin/app/cash/sqldelight/db/PgSqlCursor.kt

```diff
@@ -0,0 +1,39 @@
+package app.cash.sqldelight.db
+
+import com.benasher44.uuid.Uuid
+import kotlinx.datetime.DateTimePeriod
+import kotlinx.datetime.Instant
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.LocalDateTime
+
+/**
+ * Represents a PostgreSQL result set which can be iterated through with [next]. Initially the cursor will
+ * not point to any row, and calling [next] once will iterate to the first row.
```

Comment:
for my own edification, is there documentation somewhere on which postgres column types map to these jdbc getters?

Reply:
This documentation references interfacing with Java 8 time. 

https://jdbc.postgresql.org/documentation/head/java8-date-time.html

The docs don't explicitly call out UUID, but it's available here. https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java#L261-L266

Reply:
This would require some type casting because UUID for some reason can only be accessed through `getObject`, but that should be fine.